src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -470,37 +433,39 @@ internal static void EmitInt(this ILGenerator il, int value)
             il.Emit(c);
         }
 
-        internal static void EmitUInt(this ILGenerator il, uint value)
+        private static void EmitPrimitive(this ILGenerator il, uint value)
         {
-            il.EmitInt((int)value);
-            il.Emit(OpCodes.Conv_U4);
+            il.EmitPrimitive((int)value);
         }
 
-        internal static void EmitLong(this ILGenerator il, long value)
+        private static void EmitPrimitive(this ILGenerator il, long value)
         {
-            il.Emit(OpCodes.Ldc_I8, value);
-
-            //
-            // Now, emit convert to give the constant type information.
-            //
-            // Otherwise, it is treated as unsigned and overflow is not
-            // detected if it's used in checked ops.
-            //
-            il.Emit(OpCodes.Conv_I8);
```

Comment:
No. JIT typesystem only knows i4 and i8 integers. Unchecked Conv_I8 or Conv_U8 would be just noops here.